java-frontend/src/main/java/org/sonar/java/se/constraint/BooleanConstraint.java

```diff
@@ -46,6 +33,23 @@ public boolean isFalse() {
     return this == FALSE;
   }
 
+  @Override
+  public String valueAsString() {
+    if (this == TRUE) {
+      return "true";
+    }
+    return "false";
+  }
+
+  @Override
+  public boolean isValidWith(@Nullable Constraint constraint) {
+    return constraint == null || this == constraint;
+  }
+
+  public static boolean isBooleanConstraint(Constraint constraint) {
+    return constraint == TRUE || constraint == FALSE;
+  }
+
   public BooleanConstraint inverse() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)